### DISCUSSION TITLE
Using ruff alongside flake8-noqa

### DISCUSSION THREAD

**Discussion by Apakottur** [astral-sh/ruff#3580] [Q&A] [answered]
I am working on a project where we're using both ruff and flake8 (with the [flake8-noqa](https://github.com/plinss/flake8-noqa)) plugin.

Consider the following code:
```python
# test.py
text = "Русский"
```

Running ruff with `RUF001` enabled gives us:
```shell
ruff check --select RUF001 test.py 
test.py:1:9: RUF001 [*] String contains ambiguous unicode character `Р` (did you mean `P`?)
test.py:1:10: RUF001 [*] String contains ambiguous unicode character `у` (did you mean `y`?)
test.py:1:11: RUF001 [*] String contains ambiguous unicode character `с` (did you mean `c`?)
test.py:1:12: RUF001 [*] String contains ambiguous unicode character `с` (did you mean `c`?)
Found 4 errors.
```

If I attempt to silence the warning by modifying the file:
```python
# test.py
text = "Русский"  # noqa: RUF001
```

Then running flake8 with the flake8-noqa plugin gives us:
```shell
flake8 test.py
test.py:1:19: NQA102 "# noqa: RUF001" has no matching violations
```

It would be great if ruff could be disabled on a line with something like `# ruff-disable: RUF001` and not `# noqa` to not clash with flake8 on that matter, but perhaps there is another way to do this (without disabling the ruff rule completely, or for the entire file)?

**Comment by charliermarsh** [comment, marked as answer]
Yeah there's not a great solution to this right now. We do support the _opposite_ thing, which is that if you use Ruff's `RUF100` rule instead of `flake8-noqa`, you can mark codes that Ruff doesn't know about as "allowed" via the [`external`](https://beta.ruff.rs/docs/settings/#external) setting.

**Reply by FrostRunner** [reply]
It would be great if this rule considered the string to be good if the string is Unicode.

text = u"Русский"

**Reply by charliermarsh** [reply]
Interestingly, VS Code flags `"Русс"` but not `"Русск"` (or anything longer). Maybe they're detecting whether the string contains at least one non-"confusable" Unicode character?

**Reply by charliermarsh** [reply]
Yes, that's roughly it: https://github.com/microsoft/vscode/issues/143720#issuecomment-1048757234

**Reply by charliermarsh** [reply]
We should implement that
